crates/core/component/dex/src/component/rpc.rs

```diff
@@ -193,6 +205,77 @@ impl QueryService for Server {
         }
     }
 
+    #[instrument(skip(self, request))]
+    async fn candlestick_data(
+        &self,
+        request: tonic::Request<CandlestickDataRequest>,
+    ) -> Result<tonic::Response<CandlestickDataResponse>, Status> {
+        let state = self.storage.latest_snapshot();
+        let start_height = request.get_ref().start_height;
+        let limit = request.get_ref().limit as usize;
+        let pair: DirectedTradingPair = request
+            .get_ref()
+            .pair
+            .clone()
+            .ok_or_else(|| Status::invalid_argument("missing trading_pair"))?
+            .try_into()
+            .map_err(|_| Status::invalid_argument("invalid trading_pair"))?;
+        let prefix = state_key::candlesticks::by_pair_and_height(&pair, start_height);
+        tracing::trace!(?prefix, "searching for candlesticks from starting height");
+        let candlesticks = state
+            .nonverifiable_prefix_raw(prefix.as_bytes())
```

Comment:
I think I saw a comment about this in discord, yes you want a range query `start_height..`